Makefile

```diff
@@ -59,163 +34,230 @@ ifeq ("$(DEBUG_BUILD)", "true")
-		--build-arg PGVERSION=$(PGO_PG_VERSION) \
-		--build-arg RELVER=$(PGO_VERSION) \
-		--build-arg DOCKERBASEREGISTRY=$(DOCKERBASEREGISTRY) \
-		--build-arg PACKAGER=$(PACKAGER) \
-		--build-arg PG_FULL=$(PGO_PG_FULLVERSION) \
-		--build-arg PGVERSION=$(PGO_PG_VERSION) \
-		$(PGOROOT)
-
-%-img-buildah: %-img-build ;
-# only push to docker daemon if variable PGO_PUSH_TO_DOCKER_DAEMON is set to "true"
-ifeq ("$(IMG_PUSH_TO_DOCKER_DAEMON)", "true")
-	$(IMGCMDSUDO) buildah push $(PGO_IMAGE_PREFIX)/$*:$(PGO_IMAGE_TAG) docker-daemon:$(PGO_IMAGE_PREFIX)/$*:$(PGO_IMAGE_TAG)
-endif
-
-%-img-docker: %-img-build ;
-
-%-image: %-img-$(IMGBUILDER) ;
-
-pgo-base: pgo-base-$(IMGBUILDER)
-
-pgo-base-build: $(PGOROOT)/build/pgo-base/Dockerfile licenses
-	$(IMGCMDSTEM) \
-		-f $(PGOROOT)/build/pgo-base/Dockerfile \
-		-t $(PGO_IMAGE_PREFIX)/pgo-base:$(PGO_IMAGE_TAG) \
-		--build-arg BASE_IMAGE_OS=$(BASE_IMAGE_OS) \
-		--build-arg BASEOS=$(PGO_BASEOS) \
-		--build-arg RELVER=$(PGO_VERSION) \
-		--build-arg DOCKERBASEREGISTRY=$(DOCKERBASEREGISTRY) \
-		--build-arg PACKAGER=$(PACKAGER) \
-		--build-arg PG_FULL=$(PGO_PG_FULLVERSION) \
-		--build-arg PGVERSION=$(PGO_PG_VERSION) \
-		$(PGOROOT)
-
-pgo-base-buildah: pgo-base-build ;
-# only push to docker daemon if variable PGO_PUSH_TO_DOCKER_DAEMON is set to "true"
-ifeq ("$(IMG_PUSH_TO_DOCKER_DAEMON)", "true")
-	$(IMGCMDSUDO) buildah push $(PGO_IMAGE_PREFIX)/pgo-base:$(PGO_IMAGE_TAG) docker-daemon:$(PGO_IMAGE_PREFIX)/pgo-base:$(PGO_IMAGE_TAG)
-endif
-
-pgo-base-docker: pgo-base-build
-
-
-#======== Utility =======
+##@ Build - Images
+.PHONY: build-crunchy-postgres-exporter-image
+build-crunchy-postgres-exporter-image: ## Build the crunchy-postgres-exporter image
+build-crunchy-postgres-exporter-image: CRUNCHY_POSTGRES_EXPORTER_IMAGE_REVISION := $(shell git rev-parse HEAD)
+build-crunchy-postgres-exporter-image: CRUNCHY_POSTGRES_EXPORTER_IMAGE_TIMESTAMP := $(shell date -u +%FT%TZ)
+build-crunchy-postgres-exporter-image: build/crunchy-postgres-exporter/Dockerfile
+	$(if $(shell (echo 'buildah version 1.24'; $(word 1,$(BUILDAH_BUILD)) --version) | sort -Vc 2>&1), \
+		$(warning WARNING: old buildah does not invalidate its cache for changed labels: \
+			https://github.com/containers/buildah/issues/3517))
+	$(if $(PGO_IMAGE_TAG),,	$(error missing PGO_IMAGE_TAG))
+	$(BUILDAH_BUILD) \
+		--tag $(BUILDAH_TRANSPORT)$(PGO_IMAGE_PREFIX)/$(CRUNCHY_POSTGRES_EXPORTER_IMAGE_NAME):$(PGO_IMAGE_TAG) \
+		--build-arg PGVERSION=$(CRUNCHY_POSTGRES_EXPORTER_PG_VERSION) \
+		--label name='$(CRUNCHY_POSTGRES_EXPORTER_IMAGE_NAME)' \
+		--label build-date='$(CRUNCHY_POSTGRES_EXPORTER_IMAGE_TIMESTAMP)' \
+		--label description='$(CRUNCHY_POSTGRES_EXPORTER_DESCRIPTION)' \
+		--label maintainer='$(PGO_IMAGE_MAINTAINER)' \
+		--label summary='$(CRUNCHY_POSTGRES_EXPORTER_SUMMARY)' \
+		--label url='$(CRUNCHY_POSTGRES_EXPORTER_URL)' \
+		--label vcs-ref='$(CRUNCHY_POSTGRES_EXPORTER_IMAGE_REVISION)' \
+		--label vendor='$(PGO_IMAGE_MAINTAINER)' \
```

Comment:
🌱 An alternative to sharing the variable here might be to have separate variables that default to the same. I'm not certain, but this might allow someone to set one or both or none:

```make
PGO_IMAGE_MAINTAINER ?= Crunchy Data
CRUNCHY_POSTGRES_EXPORTER_IMAGE_MAINTAINER ?= $(PGO_IMAGE_MAINTAINER)
```